-- FreeBSD port patch for NeuralRack/makefile:
-- 1. Translate amd64 to x86_64 for UNAME_M (FreeBSD uses amd64, the build system expects x86_64)
-- 2. Use DESTDIR check instead of root user check for install directory selection
-- 3. Use PREFIX for install directories instead of hardcoded /usr paths
-- 4. Comment out root-only install guard for standalone binary
-- 5. Name the VST3 bundle directory after the compiler target (uname -m is the platform on FreeBSD)

--- NeuralRack/makefile.orig	2026-08-03 07:28:19 UTC
+++ NeuralRack/makefile
@@ -2,17 +2,17 @@
 	# check on which OS we build
 	UNAME_S := $(shell uname -s)
 	# check which architecture we build on
-	UNAME_M := $(shell uname -m)
+	UNAME_M := $(shell uname -m | sed 's/amd64/x86_64/')
 	# check which architecture we build for
 	TARGET_ARCH = UNKNOWN
 	# check if user is root
 	user = $(shell whoami)
-	ifeq ($(user),root)
-	INSTALL_DIR = /usr/lib/lv2
-	EXE_INSTALL_DIR = /usr/bin
-	CLAP_INSTAL_DIR = /usr/lib/clap
-	VST2_INSTAL_DIR = /usr/lib/vst
-	VST3_INSTALL_DIR = /usr/lib/vst3
+	ifneq ($(DESTDIR),)
+	INSTALL_DIR = $(PREFIX)/lib/lv2
+	EXE_INSTALL_DIR = $(PREFIX)/bin
+	CLAP_INSTAL_DIR = $(PREFIX)/lib/clap
+	VST2_INSTAL_DIR = $(PREFIX)/lib/vst
+	VST3_INSTALL_DIR = $(PREFIX)/lib/vst3
 	else 
 	INSTALL_DIR = ~/.lv2
 	#EXE_INSTALL_DIR = ~/bin
@@ -216,7 +216,11 @@ endif
 	VST3_INCLUDE := -I./engine/ -I./vst3/ -I./wrapper/vst3/travesty/ -I./wrapper/ -I./wrapper/vst3/ -I./plugdef/
 	VST3_SOURCES := $(VST3_DIR)NeuralRackVST3.cpp
 	ARCH = `echo $(UNAME_S) | tr A-Z a-z`
+ifeq ($(UNAME_S), FreeBSD)
+	VST3_BUNDLE := $(VST3_NAME).vst3/Contents/$(TARGET_ARCH)-$(ARCH)/
+else
 	VST3_BUNDLE := $(VST3_NAME).vst3/Contents/$(UNAME_M)-$(ARCH)/
+endif
 
 	LV2_PLUGIN := $(LV2_DIR)$(NAME)
 	LV2_GUI := $(GUI_DIR)$(NAME)
@@ -528,7 +532,7 @@ ifneq ("$(wildcard ../bin/$(EXEC_NAME)$(EXE_EXT))","")
 	@$(B_ECHO) "$(BUNDLE) skipped$(reset)"
 endif
 ifneq ("$(wildcard ../bin/$(EXEC_NAME)$(EXE_EXT))","")
-  ifeq ($(user),root)
+  #ifeq ($(user),root)
 	@$(B_ECHO) "Install  $(EXEC_NAME)$(EXE_EXT) to $(DESTDIR)$(EXE_INSTALL_DIR)/$(reset)"
 	$(QUIET)mkdir -p $(DESTDIR)$(EXE_INSTALL_DIR)/
 	$(QUIET)cp -r ../bin/$(EXEC_NAME)$(EXE_EXT) $(DESTDIR)$(EXE_INSTALL_DIR)/$(EXEC_NAME)$(EXE_EXT)
@@ -536,9 +540,8 @@ ifneq ("$(wildcard ../bin/$(EXEC_NAME)$(EXE_EXT))","")
 	$(QUIET)cp $(STANDALONE_DIR)$(NAME).desktop $(DESTDIR)$(DESKAPPS_DIR)$(NAME).desktop
 	$(QUIET)mkdir -p $(DESTDIR)$(PIXMAPS_DIR)
 	$(QUIET)cp $(STANDALONE_DIR)$(NAME).svg $(DESTDIR)$(PIXMAPS_DIR)$(NAME).svg 
-	$(QUIET)update-desktop-database || true
 	@$(B_ECHO) ". ., done$(reset)"
-  endif
+  #endif
 else
 	@$(B_ECHO) "$(EXEC_NAME)$(EXE_EXT) standalone skipped$(reset)"
 endif
